pdr_backend/aimodel/aimodel_data_factory.py

```diff
@@ -1,14 +1,14 @@
 import sys
 from typing import Tuple
```

Comment:
### Top-level comment

This PR is a great start! My feedback below is for things that still need doing for this bullet point of this issue, ie "refactor feedstr, timeframestr, pairstr, and exchangestr from tuples -> classes/objects".

You can do them as part of this PR, or you can do follow-up PR(s). 

Given that things are passing, I recommend to do it in follow-up PR(s). More tractable.

### 1. Changing **arg** modules / classes / tests:

You've created ArgFeed and got it working. Great! 
- You still need to change its module from feedstr.py -> arg_feed.py. 
- And test_feedstr.py -> test_arg_feed.py

We also want to do that for the other arg building blocks. (The github issue mentions this.)
- timeframestr.py -> arg_timeframe.py with ArgTimeframe. Change test file accordingly. There's a "Timeframe" class there too, you can either keep it there or have its own module, I'm fine either way
- pairstr.py -> arg_pair.py / ArgPair. Test file too.
- exchangestr.py -> arg_exchange.py / ArgExchange. Test file too.

### 2. Changing **subgraph** modules / classes / tests:
 - You've renamed util/feed.py::Feed to SubgraphFeed. Great!
  - You still need to change its module from feedstr.py -> subgraph_feed.py. 
  - And test_feedstr.py -> test_subgraph_feed.py

### 3. Let's rearrange the location of some files. 

We have three different groups of utility-ish (non-bot) class modules: subgraph classes, contract classes, and CLI & args classes. They're currently spread across util/ and models/
- so let's target 3 directories: "subgraph", "contract/", "cli/". How: we already have "subgraph/"; rename "models" -> "contract", and create "cli/"
- In "subgraph/", move recently-renamed subgraph*.py files from "models/". Move unit tests too ofc.
- In "cli/": move util/cli_arguments.py, util/cli_module.py, and all recently-renamed arg*.py files there. Move unit tests too.
- Then "models/" should be in good shape too. But double-check.

### 4. Q: Should I support feeds without signals, already?

Yes.

(From a Q from Calina in this PR's comments section

### Final comment

Like mentioned at top: I think this PR is good to go. Better to handle the feedback in follow-up PR(s). 

Therefore I am approving.